File: ruby/prob1.rb
=begin
problem 1
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.

=end


def arrayFunction(number, multiple1, multiple2)
  array = []
  i = 0
  number.times do
    i += 1
    array << i if i % multiple1 == 0 || i % multiple2 == 0
  end
  total = array.inject(0, :+)
return total
end

arrayFunction(1000, 3, 5)

#answer = 234168

File: prob1.m
%script to request input and perform calculation
cont == false;
number = input("Please enter the maximum number: ");
while cont == false
  if isinteger(number)
    cont == true
  else
    number = input("Please enter the maximum number: ");
  end
end
mult1 = input("Please enter the first multiple: ");
while cont == false
  if isinteger(mult1)
    cont == true;
  else
    number = input("Please enter the first multiple: ");
  end
end
mult2 = input("Please enter the second multiple: ");
while cont == false
  if isinteger(mult2)
    cont == true;
  else
    number = input("Please enter the second multiple: ");
  end
end
multarray = multiples(number, mult1, mult2);
sumtotal = sumarray(multarray);
disp ("The sum total of the all the multiples of %s or %s below %s is %s.\n", mult1, mult2, number, sumtotal);
end


function multarray = multiples(number, mult1, mult2)
% generates and returns an array containing all the multiples of mult1 and mult2 below a max number
% function call: multiples(maxnumber, multiple1, multiple2)
multarray = zeros(number, 1);
for i = 1:number
  if mod(i, mult1) == 0 || mod(i, mult2) == 0
    multarray(i) = i;
  end
end
end


function sumtotal = sumarray(array)
% returns the sum of an array
% function call: sumarray(array)
sumtotal = sum(array);
end
